### PR TITLE
Dashboard Info Panel

### PR DESCRIPTION
**Pull request by npfitz**
Epic #22826 

Dashboard Info Panel with history and editable description. Removing `DashboardHistoryModal`, `DashboardDetailsModel` and related tests will follow this PR. A few notes:
- Sizing of the side panel feels a little strange to me. if you have a "short" dashboard, but lots of revisions to show, it's a full screen scroll.
- The revisions aren't quite as nice as they are with questions. Might be something to look into.
chrome_fkDez2zluI

### CODE REVIEW

File: frontend/src/metabase/core/components/EditableText/EditableText.tsx [hunk 1]
```diff
@@ -41,6 +42,11 @@ const EditableText = forwardRef(function EditableText(
   const [submitValue, setSubmitValue] = useState(initialValue ?? "");
   const displayValue = inputValue ? inputValue : placeholder;
 
+  useEffect(() => {
+    setInputValue(initialValue ?? "");
+    setSubmitValue(initialValue ?? "");
+  }, [initialValue, setInputValue, setSubmitValue]);
+
```
Comment (npfitz) [inline frontend/src/metabase/core/components/EditableText/EditableText.tsx:49]: Done to support revisions on Dashboards. On Questions we effectively reload the whole query_builder component on revert, but I wasn't sure we would want to do that with Dashboards.

File: frontend/src/metabase-types/api/dashboard.ts [hunk 2]
```diff
@@ -8,6 +8,7 @@ import { CardId, SavedCard } from "metabase-types/types/Card";
 export interface Dashboard {
   id: number;
   name: string;
+  description: string;
```
Comment (kulyk) [inline frontend/src/metabase-types/api/dashboard.ts:11]: Description can also be `null`

File: frontend/src/metabase/dashboard/components/DashboardInfoSidebar.tsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+  );
+
+  return (
+    <DashboardInfoSidebarRoot>
+      <DashboardDescriptionEditbaleText
+        initialValue={dashboard.description}
+        isDisabled={!dashboard.can_write}
+        onChange={handleDescriptionChange}
+        isMultiline
+        placeholder="Add a helpful description. You'll thank me later"
```
Comment (kulyk) [inline frontend/src/metabase/dashboard/components/DashboardInfoSidebar.tsx:64]: Missing i18n

File: frontend/src/metabase/core/components/EditableText/EditableText.tsx (hunk 1 above)
Comment (npfitz) [reply frontend/src/metabase/core/components/EditableText/EditableText.tsx:49]: That's actually a great idea... I've known about using keys in maps, but never thought to just add a key to an attribute to have it re-mount when needed. I've just added the whole description to the key. Do you think that's sufficient? Or should we run a checksum on the string to keep the length down?

File: frontend/src/metabase/dashboard/components/DashboardInfoSidebar.tsx [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+  revertToRevision,
+};
+
+export default _.compose(
+  Revision.loadList({
+    query: (state: State, props: DashboardInfoSidebarProps) => ({
+      model_type: "dashboard",
+      model_id: props.dashboard.id,
+    }),
+    wrapped: true,
```
Comment (ranquild) [inline frontend/src/metabase/dashboard/components/DashboardInfoSidebar.tsx:92]: btw, is this actually used? `wrapped: true` adds `selectors` and `objectActions` from the entity definition directly to the entity object. I don't see it's used here

**Review by ranquild — APPROVED** [review]

File: frontend/src/metabase/dashboard/components/DashboardInfoSidebar.tsx (hunk 4 above)
Comment (npfitz) [reply frontend/src/metabase/dashboard/components/DashboardInfoSidebar.tsx:92]: Ahh, It is not. I just took that example from somewhere else. Thanks for the heads up!

Comment (npfitz) [reply frontend/src/metabase/dashboard/components/DashboardInfoSidebar.tsx:92]: Actually, yes it is. We pass that revision object to an action and call `revert()` on it, which is an object action

**Pull request merged**